rsconnect/main.py

```diff
@@ -869,28 +867,19 @@ def deploy_html(
 ):
     set_verbosity(verbose)
 
-    entrypoint = entrypoint or infer_entrypoint(path, "text/html")
-    manifest = make_html_manifest(entrypoint)
-
     with cli_feedback("Checking arguments"):
         connect_server = _validate_deploy_to_args(name, server, api_key, insecure, cacert)
-        manifest = validate_manifest_file(manifest)
-        app_store = AppStore(manifest)
+        app_store = AppStore(path)
 
         (
             app_id,
             deployment_name,
             title,
             default_title,
             app_mode,
-            package_manager,
-        ) = gather_basic_deployment_info_from_manifest(connect_server, app_store, manifest, new, app_id, title)
+        ) = gather_basic_deployment_info_for_notebook(connect_server, app_store, path, new, app_id, title, True)
```

Comment:
`gather_basic_deployment_info_for_notebook` will raise an exception if the give path isn't an `.ipynb` file. You might need a variant of that function specific to static deployments.